lib/private/Files/ObjectStore/S3ObjectTrait.php

```diff
@@ -87,25 +95,34 @@ public function writeObject($urn, $stream) {
 			$count += $read;
 		});
 
-		$uploader = new MultipartUploader($this->getConnection(), $countStream, [
+		$s3params = [
 			'bucket' => $this->bucket,
 			'key' => $urn,
 			'part_size' => $this->uploadPartSize,
-		]);
+		];
+		$s3params['params'] = $this->getSseKmsPutParameters();
+		$uploader = new MultipartUploader($this->getConnection(), $countStream, $s3params);
 
 		try {
 			$uploader->upload();
 		} catch (S3MultipartUploadException $e) {
 			// This is an empty file so just touch it then
 			if ($count === 0 && feof($countStream)) {
-				$uploader = new ObjectUploader($this->getConnection(), $this->bucket, $urn, '');
+				$s3params = [];
+				$s3params['params'] = $this->getSseKmsPutParameters();
```

Comment:
This can be directly initialised :) 

```suggestion
				$s3params = [
					'params' =  $this->getSseKmsPutParameters();
				];
```